### PR TITLE
Additional parameters in some functions - new functions #460

### PR DESCRIPTION
**Pull request by andreatangherloni**

### CODE REVIEW

**Comment by andreatangherloni** [comment]
I tested the code, now it is possible to calculate more components for the tSNE embedding method

**Comment by falexwolf** [comment]
These are a lot of different things in one PR.

I'm providing some comments in the code.

File: scanpy/utils.py [hunk 1]
```diff
@@ -667,8 +667,22 @@ def select_groups(adata, groups_order_subset='all', key='groups'):
     if groups_order_subset != 'all':
         groups_ids = []
         for name in groups_order_subset:
-            groups_ids.append(
-                np.where(adata.obs[key].cat.categories.values == name)[0][0])
+            # deal with case that all group names were originally integers,
```
Comment (falexwolf) [inline scanpy/utils.py:670]: All of this is a lot of code that doesn't handle Scanpy's deprecation of non-string categories in a good way. So, I don't like to merge this.

Instead, we should print a warning if someone operates with non-string categories. If you just make it a warning, here, `logg.warn`, that's fine.

**Comment by falexwolf** [comment]
OK, wasn't too much:
- happy to merge a PR around tSNE
- there is a PR on the wilcoxon issue, here: https://github.com/theislab/scanpy/pull/519 I know it came later than yours (the result of my sickness and absence here...), but I guess @a-munoz-rojas, who contributed a lot to rank_genes_groups already should make these changes

**Comment by andreatangherloni** [comment]
@falexwolf many thanks for the work you are doing.

**Comment by falexwolf** [comment]
Thank you for contributing, @andrea-tango!

**Pull request closed without merging**